cpp/ql/lib/semmle/code/cpp/models/interfaces/FormattingFunction.qll

```diff
@@ -170,6 +170,16 @@ abstract class FormattingFunction extends ArrayFunction, TaintFunction {
       output.isParameterDeref(this.getOutputParameterIndex(_))
     )
   }
+
+  final override predicate isPartialWrite(FunctionOutput output) {
+    exists(int outputParameterIndex |
+      output.isParameterDeref(outputParameterIndex) and
```

Comment:
Given where you added this, this presumably also leads more flow on the first argument of functions like `sprintf`. Is that correct? If so, is that what we want?

Reply:
The `this.getOutputParameterIndex(true)` conjunct only holds for formatting functions which specify that the output parameter is a stream, and `sprintf` [implements this predicate with](https://github.com/github/codeql/blob/4684ac1ed97ed4096afed0f1200d2114635569ad/cpp/ql/lib/semmle/code/cpp/models/implementations/Printf.qll#L88) `isStream = false`